wallet/reference/non-evm-apis/starknet-snap-api.md

```diff
@@ -0,0 +1,830 @@
+    request: {
+      method: "starkNet_estimateFee",
+      params: {
+        contractAddress: "0x5B38Da6a701c568545dCfcB03FcB875f56beddC4",
+        contractFuncName: "transfer",
+        contractCallData: "0x456...,0x789...,100",
+        senderAddress: "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+        chainId: "0x1"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+{
+  "suggestedMaxFee": "1000000000000000",
+  "overallFee": "900000000000000",
+  "gasConsumed": "1000000",
+  "gasPrice": "1000000000",
+  "unit": "wei",
+  "includeDeploy": false
+}
+```
+
+</TabItem>
+</Tabs>
+
+## `starkNet_extractPublicKey`
+
+:::note
+This method is integrated into `get-starknet`.
+:::
+
+Extracts the public key from a Starknet account address.
+
+### Parameters
+
+- `userAddress`: `string` - Address of the account contract.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Goerli testnet.
+
+### Returns
+
+The public key of the given account address (can be different from the actual signer).
+
+### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await window.ethereum.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_extractPublicKey",
+      params: {
+        userAddress: "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+        chainId: "0x1"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+"0x04bfcab3b7ca7e8b3f3b62b2f7f77e9e4b68080bbf8f0f4a1c8f890864d2c7c1d3c45d8b2e3f5f1c27dfeea4c2f5733e90bfc7484e2a690aa9b8ac4559d2e6a8d7"
+```
+
+</TabItem>
+</Tabs>
+
+## `starkNet_getErc20TokenBalance`
+
+Gets the user's current balance of an ERC-20 token.
+
+### Parameters
+
+- `tokenAddress`: `string` - Address of the ERC-20 token contract.
+- `userAddress`: `string` - Address of the user account.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Goerli testnet.
+
+### Returns
+
+The token balance in hexadecimal.
+
+### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await window.ethereum.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_getErc20TokenBalance",
+      params: {
+        tokenAddress: "0x5B38Da6a701c568545dCfcB03FcB875f56beddC4",
+        userAddress: "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+        chainId: "0x1"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+"0x3e8"
+```
+
+</TabItem>
+</Tabs>
+
+## `starkNet_getStoredUserAccounts`
+
+Gets a list of stored user accounts that are either initialized or initializing.
+
+### Parameters
+
+`chainId`: `string` - (Optional) ID of the target Starknet network.
+The default is the Starknet Goerli testnet.
+
+### Returns
+
+The list of the stored user accounts.
```

Comment:
There are two new optional return parameters in the account: `upgradeRequired` and `deployRequired`. 

User accounts type is as follow : 

```typescript
export type AccContract = {
  addressSalt: string;
  publicKey: string; // in hex
  address: string; // in hex
  addressIndex: number;
  derivationPath: string;
  deployTxnHash: string; // in hex
  chainId: string; // in hex
  upgradeRequired?: boolean; // whether the account requires an upgrade to Cairo 1
  deployRequired?: boolean; // whether the account requires to be deployed
};
```

Reply:
@khanti42 I have questions about this one (raised in chat)

Reply:
@khanti42 i think we dont need to mention this method, as it is a method to return the deployed + stored accounts for a single network, consider some edge case for Cairo Contract, it is better not have any starkNet_getStoredXXXX method mentioned in the page

cc @joaniefromtheblock

Reply:
this has been removed